### PR TITLE
Refactor/sync data

### PR DESCRIPTION
**Pull request by consolethinks**
Refactors syncDataToFileserver:
 - redirect progress report to optional io.Writer parameter
 - remove logging

### CODE REVIEW

File: datasetIngestor/syncDataToFileserver_unix.go [hunk 1]
```diff
@@ -1,39 +1,42 @@
...
 	"strings"
+
 	version "github.com/mcuadros/go-version"
-	"regexp"
 )
 
 // functionality needed for "de-central" data
-func SyncDataToFileserver(datasetId string, user map[string]string, RSYNCServer string, sourceFolder string, absFileListing string) (err error) {
+// copies data from a local machine to a fileserver, uses RSync underneath
+func SyncLocalDataToFileserver(datasetId string, user map[string]string, RSYNCServer string, sourceFolder string, absFileListing string, cmdOutput io.Writer) (err error) {
```
Comment (minottic) [inline datasetIngestor/syncDataToFileserver_unix.go:19]: what's the reason for this extra arg, cmdOutput?

**Review by minottic — COMMENTED** [review]
minor comment + I would remove the commented logs

File: datasetIngestor/syncDataToFileserver_unix.go (hunk 1 above)
Comment (minottic) [reply datasetIngestor/syncDataToFileserver_unix.go:19]: is that used now?

Comment (consolethinks) [reply datasetIngestor/syncDataToFileserver_unix.go:19]: It is used for outputting the rsync progress to the stdout, when using the datasetIngest command with the copy flag

Comment (minottic) [reply datasetIngestor/syncDataToFileserver_unix.go:19]: but the file one is not used if I am not mistaken. Anyway, it doesn't harm

**Review by minottic — APPROVED** [review]

**Pull request merged**